src/mcts/search.cc

```diff
@@ -1435,6 +1435,12 @@ void SearchWorker::FetchSingleNodeResult(NodeToProcess* node_to_process,
                                 edge.GetMove().as_nn_index(
                                     node_to_process->probability_transform)));
   }
+  // Intermediate array to store values when processing policy.
+  // According to a lichess developer post:
+  // https://lichess.org/blog/Wqa7GiAAAOIpBLoY/
+  //     developer-update-275-improved-game-compression
+  // there are never more than 256 valid legal moves in any legal position.
+  float intermediate[256];
```

Comment:
I suggest to use std::array<> instead, it basically has no drawbacks to legacy arrays and only benefits.

Reply:
Thanks, it is changed now.
```
===========================
Total time (ms) : 340358
Nodes searched  : 70115712
Nodes/second    : 206005
```

Reply:
How are you testing? I get a compiler error with clang:
```
FAILED: lc0@exe/src_mcts_search.cc.o 

../../src/mcts/search.cc:1440:26: error: implicit instantiation of undefined template 'std::__1::array<float, 256>'
  std::array<float, 256> intermediate;
                         ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/__tuple:219:64: note: template is declared here
template <class _Tp, size_t _Size> struct _LIBCPP_TEMPLATE_VIS array;
                                                               ^
1 error generated.
```

Reply:
I'm using clang too, works for me just fine.
```
C++ compiler for the host machine: clang++-6.0 (clang 6.0.1-11 "clang version 6.0.1-11 (tags/RELEASE_601/final)")
C++ linker for the host machine: clang++-6.0 ld.bfd 2.33
```